packages/xdl/src/Web.js

```diff
@@ -1,15 +1,62 @@
 import opn from 'opn';
-
+import chalk from 'chalk';
 import Logger from './Logger';
 import * as UrlUtils from './UrlUtils';
+import { readConfigJsonAsync } from './project/ProjectUtils';
 
-export async function openProjectAsync(projectRoot) {
+export async function openProjectAsync(projectRoot, options) {
+  const hasWebSupport = await hasWebSupportAsync(projectRoot);
+  if (!hasWebSupport) {
+    logWebSetup();
+    return { success: false };
+  }
   try {
-    let url = await UrlUtils.constructWebAppUrlAsync(projectRoot);
+    let url = await UrlUtils.constructWebAppUrlAsync(projectRoot, options);
     opn(url, { wait: false });
     return { success: true, url };
   } catch (e) {
     Logger.global.error(`Couldn't start project on web: ${e.message}`);
     return { success: false, error: e };
   }
 }
+
+export function logWebSetup() {
+  Logger.global.error(getWebSetupLogs());
+}
+
+export async function hasWebSupportAsync(projectRoot) {
+  const { exp } = await readConfigJsonAsync(projectRoot);
+  const isWebConfigured = exp.platforms.includes('web');
+  return isWebConfigured;
+}
+
+function getWebSetupLogs() {
+  const appJsonRules = chalk.white(
+    `
+  ${chalk.whiteBright.bold(`app.json`)}
+  {
+    "packages": [
```

Comment:
```suggestion
    "platforms": [
```
(The correct field is `platforms`, not `packages`.)